src/core/notifier/event/event.go

```diff
@@ -0,0 +1,121 @@
+package event
+
+import (
+	"time"
+
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/notifier"
+	"github.com/goharbor/harbor/src/core/notifier/model"
+)
+
+// Event to publish
+type Event struct {
```

Comment:
Generally this refactor is slick! Thanks!
But why not put general info like `OccurAt`,`Operator`  in the `Event struct` rather than `Event.Data`?

Reply:
We'll delay the change.